src/sys/resource.rs

```diff
@@ -0,0 +1,48 @@
+use std::mem;
+
+use libc::{self, c_int};
+
+use {Errno, Result};
+
+#[repr(i32)]
+pub enum Resource {
```

Comment:
yeah I thought of this kind of approach but I don't know yet how I feel about going so far from the C API.

In related news: I just created https://github.com/nix-rust/rfcs and will create RFC 0000 soon (probably tomorrow) to bootstrap the process. Then we'll have a good place to discuss stuff like this.